pilot/cmd/pilot-agent/main.go

```diff
@@ -223,6 +224,28 @@ var (
 			if err != nil {
 				return err
 			}
+			tokenManager := tokenmanager.CreateTokenManager(tokenManagerPlugin,
+				tokenmanager.Config{CredFetcher: secOpts.CredFetcher, TrustDomain: secOpts.TrustDomain})
+			secOpts.TokenManager = tokenManager
+
+			// If security token service (STS) port is not zero, start STS server and
+			// listen on STS port for STS requests. For STS, see
+			// https://tools.ietf.org/html/draft-ietf-oauth-token-exchange-16.
+			if stsPort > 0 {
```

Comment:
Sorry, I still don't get it why this is added here and what STS has to do with XDS auth. If this is for stackdriver - move to separate PR.

Reply:
Or is this code just moved with same semantics ? In any case - add a comment, 'STS is used for stackdriver or other Envoy services using google gRPC'

Reply:
This code is in existing implementation for the sts server used by the token exchange for telemetry.
sts server has nothing to do with XDS auth but is used for the token exchange for stacker driver.

Reply:
Added the comment 'STS is used for stackdriver or other Envoy services using google gRPC'.